Mathlib/Geometry/Manifold/InteriorBoundary.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2023 Michael Rothgang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Michael Rothgang
+-/
+
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Interior and boundary of a manifold
+Define the interior and boundary of a manifold.
+
+## Main definitions
+- **IsInteriorPoint x**: `p ∈ M` is an interior point if, for `φ` being the preferred chart at `x`,
+ `φ x` is an interior point of `φ.target`.
+- **IsBoundaryPoint x**: `p ∈ M` is a boundary point if, `(extChartAt I x) x ∈ frontier (range I)`.
+- **interior I M** is the **interior** of `M`, the set of its interior points.
+- **boundary I M** is the **boundary** of `M`, the set of its boundary points.
+
+## Main results
+- `univ_eq_interior_union_boundary`: `M` is the union of its interior and boundary
+- `interior_boundary_disjoint`: interior and boundary of `M` are disjoint
+- if `M` is boundaryless, every point is an interior point
+
+## Tags
+manifold, interior, boundary
+
+## TODO
+- `x` is an interior point iff *any* chart around `x` maps it to `interior (range I)`;
+similarly for the boundary.
+- the interior of `M` is open, hence the boundary is closed (and nowhere dense)
+  In finite dimensions, this requires e.g. the homology of spheres.
+- the interior of `M` is a smooth manifold without boundary
+- `boundary M` is a smooth submanifold (possibly with boundary and corners):
+follows from the corresponding statement for the model with corners `I`;
+this requires a definition of submanifolds
+- if `M` is finite-dimensional, its boundary has measure zero
+
+-/
+
+open Set
+
+-- Let `M` be a manifold with corners over the pair `(E, H)`.
+variable {𝕜 : Type*} [NontriviallyNormedField 𝕜]
+  {E : Type*} [NormedAddCommGroup E] [NormedSpace 𝕜 E]
+  {H : Type*} [TopologicalSpace H] {I : ModelWithCorners 𝕜 E H}
+  {M : Type*} [TopologicalSpace M] [ChartedSpace H M] [HasGroupoid M (contDiffGroupoid 0 I)]
+
+/-- `p ∈ M` is an interior point of a manifold `M` iff
+for `φ` being the preferred chart at `x`, `φ x` is an interior point of `φ.target`. -/
+def ModelWithCorners.IsInteriorPoint (x : M) := extChartAt I x x ∈ interior (range I)
```

Comment:
`I` should be explicit in this definition: otherwise, when you call `IsInteriorPoint x`, there is no way Lean can guess which model space `H` you are using.